### ISSUE TITLE
Ueli freezes often

### ISSUE THREAD

**Issue by HermasciousLevander** [oliverschwendener/ueli#701]
I have no idea what is causing this, but for the better part of forever, Ueli simply freezes. Today I've had it freeze somewhere between five or ten times, the system has not been active for that many hours in total, around five to maybe seven. It has also closed for no reason, which I assume is it crashing, that's new at least. Freezing comes and goes, but sometimes it freezes right after starting. Debug logs show nothing of note, simply that it has been been `Successfully refreshed all indexes` and occasionally looked for updates.

The system is put to sleep after around 7-8 hours of usage, and reactivated again the morning after. This is a work machine, which is why I'm to lazy to shut it off completely. Hibernation does mess with the system somewhat and I've seen HDDs not being able to drag themselves back to life from it back in the day, may be something similar here. Regardless, it is somewhat strange that it freezes so often, but eh ¯\\_(ツ)\_/¯

Currently monitoring all ueli processes to see what it is doing and when/how it stops working.

---

## Ueli information:

Ueli: `8.12.0`
Electron: `11.2.3`
Node: `12.18.3`
V8: `9.7.220.31-electron.0`

## Operating System information:

Windows 10: `Enterprise (Win32NT 10.0.18363.0)`
Windows version: `1909`
Build: `18363.1440`

## Machine information:

Machine: `Lenovo ThinkPad L480`
Modelnumber: `20LTS10Q00`
BIOS version: `R0QET63W (1.40 )`
Processor: `Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz`
Processor caption: `Intel64 Family 6 Model 142 Stepping 10`
Processor socket: `U3E1`

**Comment by oliverschwendener** [comment]
Did you change any settings?

**Comment by HermasciousLevander** [comment]
Ah, apologies, forgot to mention that.

I've disabled most of the features, only the color converter, shortcuts and workflow is active. Here's [my config](https://pastebin.com/swjek8RB), though I've removed the shortcuts and workflow manually. Pretty sure I didn't mess up the config, but it's mostly because it's not really necessary considering the amount of them I have.

**Comment by HermasciousLevander** [comment]
Do you have anything I can test to see if it fixes it? Ueli is quite nice but it is pretty annoying to have to quit and start it so often.

**Comment by oliverschwendener** [comment]
To be honest I have no clue what it could be. 🤔 What I would suggest is that you disable all plugins and enable one after the other to see if a particular plugin is causing the issue.

**Comment by HermasciousLevander** [comment]
I'll give that a try. I'll add some more information to the OP post while I do.

**Comment by HermasciousLevander** [comment]
Alright, so it has frozen again now. This is with color converter, shortcuts and workflow still active though, it's mostly just to see if I could capture what it was freezing on.

![image](https://user-images.githubusercontent.com/75787425/116542560-09333b00-a8ed-11eb-9445-f07a3a634b35.png)

I killed cmd with PID 10484, which made no difference. conhost with PID 25784 was gone when I went to look for it next. Ueli was still unresponsive. I then killed mstsc with PID 23980 to end the remote desktop session, at which point Ueli started working. It _might_ have been mstsc, or it might have been cmd but it hung a while after I killed it, not sure. Leaning towards mstsc though.

I'm seeing some buffer_overflow in the main ueli instance with PID 5040, specifically here HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\ICM\RegisteredProfiles\camp with a length of 12. Though if that means anything to you, I have no idea.

![image](https://user-images.githubusercontent.com/75787425/116544617-aa22f580-a8ef-11eb-82ad-507284689c1d.png)

If I was to make an assumption, it'd be that when Ueli spawns a process, it doesn't properly disconnect it from its own subtree/make it its own process given some edge-cases like mine where the spawned process never really closes because its an active remote desktop session and this eventually causes Ueli to freeze.

How about you? Any ideas on how this can be solved/anything else I can test?
